compiler/modules/front.syntax/ast.bal

```diff
@@ -477,7 +477,7 @@ public type MappingTypeDesc record {|
     Position startPos;
     Position endPos;
     FieldDesc[] fields;
-    TypeDesc? rest;
+    TypeDesc|boolean rest; // true if inclusive record type false if rest is nil
```

Comment:
The comment is inaccurate.
`true` means the record rest comes implicitly from the use of an inclusive-record-type-descriptor. `false` means there is no record-rest.

Actually I think on reflection it is better for this to be `TypeDesc|INCLUSIVE_RECORD_TYPE_DESC|()`, with

```
public const INCLUSIVE_RECORD_TYPE_DESC = true;
```

Reply:
fixed